adcc/AdcMatrix.py

```diff
@@ -112,19 +155,72 @@ def __init__(self, method, hf_or_mp, block_orders=None, intermediates=None):
         # Build the blocks and diagonals
         with self.timer.record("build"):
             variant = None
-            if method.is_core_valence_separated:
+            if self.is_core_valence_separated:
                 variant = "cvs"
             self.blocks_ph = {  # TODO Rename to self.block in 0.16.0
                 block: ppmatrix.block(self.ground_state, block.split("_"),
                                       order=order, intermediates=self.intermediates,
                                       variant=variant)
-                for block, order in block_orders.items() if order is not None
+                for block, order in self.block_orders.items() if order is not None
             }
             self.__diagonal = sum(bl.diagonal for bl in self.blocks_ph.values()
                                   if bl.diagonal)
             self.__diagonal.evaluate()
             self.__init_space_data(self.__diagonal)
 
+    def __iadd__(self, other):
+        """In-place addition of an :py:class:`AdcExtraTerm`
+
+        Parameters
+        ----------
+        other : AdcExtraTerm
+            the extra term to be added
+        """
+        if isinstance(other, list):
+            for k in other:
+                self += k
+            return self
+        if not isinstance(other, AdcExtraTerm):
+            raise TypeError("Can only add AdcExtra term or"
+                            " lists thereof to AdcMatrix.")
```

Comment:
That's not the right way to do this in python. if the types don't match throw NotImplemented. See docs on python operator overloading.

Reply:
Done.